src/DevBetterWeb.Core/Entities/BookCategory.cs

```diff
@@ -26,24 +26,37 @@ public void ExcludeMembers(List<int> excludedMembersIds)
 	  foreach (var book in Books)
 	  {
 		  var newMembers = book.MembersWhoHaveRead!.Where(memberWhoHaveRead => !excludedMembersIds.Contains(memberWhoHaveRead.Id)).ToList();
-		  book.MembersWhoHaveRead = newMembers;
-	  }
+		  book.MembersWhoHaveRead = newMembers.OrderBy(x => x.BooksRank).ToList();
+		}
   }
+	public void AddMembersRole(List<int> excludedMembersIds)
+	{
+		foreach (var book in Books.Where(book => book.MembersWhoHaveRead != null))
+		{
+			foreach(var memberWhoHaveRead in book.MembersWhoHaveRead!.Where(memberWhoHaveRead => !excludedMembersIds.Contains(memberWhoHaveRead.Id)))
+			{
+				memberWhoHaveRead.SetRoleName(AuthConstants.Roles.MEMBERS);
+			}
+		}
+	}
+	
 
 	public static void CalcAndSetCategoriesBooksRank(RankingService<int> rankingService, List<BookCategory> bookCategories)
   {
 	  foreach (var bookCategory in bookCategories)
 	  {
 		  bookCategory.CalcAndSetBooksRank(rankingService);
-	  }
-  }
+			bookCategory.Books = bookCategory.Books.OrderBy(x => x.Rank).ToList();
+		}
+	}
 
   public static void CalcAndSetMemberCategoriesMembersRank(RankingService<int> rankingService, List<BookCategory> bookCategories)
   {
 	  foreach (var bookCategory in bookCategories)
 	  {
 		  bookCategory.CalcAndSetMembersRank(rankingService);
-	  }
+			bookCategory.Books = bookCategory.Books.OrderBy(x => x.Rank).ToList();
```

Comment:
Same comment as above. The ordering doesn't need to happen in the entity - it can happen in the viewmodel or API or somewhere closer to the presentation layer.